### ISSUE TITLE
Update parser in accordance with changes made at hydra-python-core

### ISSUE THREAD

**Issue by vddesai1871** [HTTP-APIs/hydra-openapi-parser#19]
### I'm submitting a 
- [ ] bug report.
- [x] feature request.

### Current Behaviour:

The current parser uses old Hydra terms which are replaced since long, as we have updated hydra-python-core with developments made at Hydra-CG spec, outdated parser will create a HydraDoc incompatible with hydra-python-core and hydrus.
### Expected Behaviour:

The parser should be updated with respect to changes made at hydra-python-core and thus on parsing it should create a HydraDoc compatible with latest Hydra Spec.
### Do you want to work on this issue?

YES

**Comment by roufiq** [comment]
no follow up?

**Comment by chrizandr** [comment]
A lot of changes in both hydra-spec and the openapi spec have been made since this project was last updated. This issue is only regarding updates with hydra-python-core, but the broader issue of updating in accordance to the spec needs to be done before this issue can be taken up.

**Comment by farazkhanfk7** [comment]
If no one is working on it I would like to take this one.

**Comment by amalthundiyil** [comment]
@chrizandr Would it be acceptable to resolve this issue by fixing the parser at its current state first and then addressing the "broader issue" by creating a separate issue?

**Comment by farazkhanfk7** [comment]
@amal-thundiyil  The current version of hydra-openapi-parser is outdated. It uses old Hydra terms which are
no longer in use. Also, hydra-python-core has gone through lots of developments since the
last time parser was updated, as a result it throws error and returns an incompatible
HydraDoc. Many functions in hydra-openapi-parser are missing parameters. The parser should be
updated according to the changes in hydra-python-core and Hydra Specs. 
And yes, you can create a separate issue for that as well.
